modules/currency-l0/src/main/scala/org/tessellation/currency/l0/CurrencyL0App.scala

```diff
@@ -58,8 +58,9 @@ abstract class CurrencyL0App(
 
     for {
       queues <- Queues.make[IO](sdkQueues).asResource
-      p2pClient = P2PClient.make[IO](sdkP2PClient, sdkResources.client, sdkServices.session)
       storages <- Storages.make[IO](sdkStorages, cfg.snapshot, method.globalL0Peer).asResource
+      l0Peers <- storages.globalL0Cluster.getPeers.map(_.toSortedSet).asResource
```

Comment:
this `get` should be directly before fetching trust. L0 peers may change every snapshot or few snapshots. the way it's implemented right now you pass static list to `p2pClient` and it's never updated again through discovery. so after some time you may have outdated list of l0 peers and you won't be able to fetch trust at all

Reply:
Why in such unusual way? The service should get L0ClusterStorage or peer storage as an argument, there is no need to pass an additional method

Reply:
Per our offline discussion, `L0ClusterStorage` is being passed to `L0TrustClient`, where the latest `L0Peer`s are retrieve, `RandomPeerSelect` is created, and a peer selected before the call is made.